packages/run-protocol/src/vpool-xyk-amm/addPool.js

```diff
@@ -12,50 +13,92 @@ const { details: X } = assert;
  * @param {ZCF} zcf
  * @param {(Brand) => boolean} isInSecondaries
  * @param {WeakStore<Brand,ZCFMint>} brandToLiquidityMint
+ * @param {WeakStore<Brand,ERef<Issuer>>} brandToLiquidityIssuer
  * @param {() => (secondaryBrand: Brand) => Promise<void>} getAddIssuerToReserve
  */
 export const makeAddIssuer = (
   zcf,
   isInSecondaries,
   brandToLiquidityMint,
+  brandToLiquidityIssuer,
   getAddIssuerToReserve,
 ) => {
   /**
+   * Add a new issuer. If we previously received a request for the same issuer,
+   * we'll return a promise or the issuer.
```

Comment:
did you mean
```suggestion
   * we'll return a promise for the issuer.
```
?

Reply:
I actually meant that we'll return either the issuer itself or a promise for the issuer, depending.  On line 58, we return the issuer, while on line 46, we return a Promise. Since the method is `async`, the caller will always get a promise, but sometimes it's already resolved, and other times it'll be resolved later.